src/main/java/com/sforce/cd/apexUnit/client/utils/ApexClassFetcherUtils.java

```diff
@@ -170,7 +170,15 @@
 		if (regex != null && !regex.equals(" ")) {
 			LOG.info("Using regex: \"" + regex + "\" to fetch apex classes");
 			// construct the query
+			// if a period is specified, then the text before the period will be interpreted as the namespace
+			String[] splitByPeriod = regex.split("\\.", 2);
 			String namespace = null;
+
+			if (splitByPeriod.length == 2) {
```

Comment:
Some thoughts:
1. What if there is no period(.) ?  namespace should be empty ?
2. If length is 1(example: ".some_regex" or "something."; I feel we should throw an error because its hard to guess whether user intended to provide the namespace or the regex
3. if length > 3 . Error out
@awillia2 @vamshi-sfdc